p2p/peer_set_test.go

```diff
@@ -65,3 +70,50 @@ func TestAddRemoveMany(t *testing.T) {
 		}
 	}
 }
+
+func TestPeerSetAddDuplicate(t *testing.T) {
+	t.Parallel()
+	peerSet := NewPeerSet()
+	peer := randPeer()
+	if err := peerSet.Add(peer); err != nil {
+		t.Fatalf("Failed to add new peer: %v", err)
+	}
+
+	var wg sync.WaitGroup
+	for i := 0; i < 10; i++ {
+		// Add them asynchronously to test the
+		// concurrent guarantees of our APIs.
+		wg.Add(1)
+		go func(i int) {
+			defer wg.Done()
+			got, want := peerSet.Add(peer), ErrSwitchDuplicatePeer
+			assert.Equal(t, got, want, "#%d: got=%v want=%v", i, got, want)
+		}(i)
+	}
+	wg.Wait()
+}
+
+func TestPeerSetGet(t *testing.T) {
+	t.Parallel()
+	peerSet := NewPeerSet()
+	peer := randPeer()
+	var nilPeer *Peer
+	assert.Equal(t, peerSet.Get(peer.Key), nilPeer, "expecting a nil lookup, before .Add")
```

Comment:
shouldn't `PeerSet.Get` return `nil`?

Reply:
It returns a typed nil so (*PeerSet)(nil) , which assert.Equal trips out over unfortunately.

Reply:
what about `assert.Nil`?

Reply:
Aha, good plan, thanks for letting me know. Let me try that.